src/components/Editor/index.js

```diff
@@ -275,7 +275,10 @@ class Editor extends PureComponent<Props, State> {
 
     if (e.shiftKey) {
       this.toggleConditionalPanel(line);
+    } else if (!conditionalPanelLine) {
+      this.props.toggleBreakpoint(line);
     } else {
+      this.toggleConditionalPanel(line);
       this.props.toggleBreakpoint(line);
     }
```

Comment:
I think this might be a good way to do it:

```js
   if (e.shiftKey) {
       this.toggleConditionalPanel(line);
     } else {
       if (conditionalPanelLine) {
          this.toggleConditionalPanel(line);
       }
       this.props.toggleBreakpoint(line);
     }
```

Perhaps an even better way to do it would be

```js
   if (e.shiftKey) {
       this.toggleConditionalPanel(line);
     } else {
       this.props.toggleBreakpoint(line);
     }
```
and then reset the conditional panel in the redux action

Reply:
Your first suggested attempt is forbidden by some EsLint rule and I really wouldn't move it into the toggleBreakpoint  action as it makes things really unclear

Reply:
We added all of this kind of logic into actions on another project I worked on and things became really nasty over time. We refactored all of that just to get back to a healthy source.
Just think to how this action should be named to reflect its purpose: `addOrRemoveBreakpointAndCloseConditionalPanelIfOpen`
This sounds to me this action would do way to much...

Reply:
> addOrRemoveBreakpointAndCloseConditionalPanelIfOpen

lol

It's true, that we have been moving some of our component complexity to the actions. It's quite possible that we will find that they become too complicated as well.

Another interesting location for this would be the reducers, which could listen for ADD_BREAKPOINT, and reset the conditional pane.

Either way, we are splitting hairs here and i'm happy leaving the logic in the component.

Reply:
Continuing with splitting hairs: it seems like `toggleConditionalPanel` is doing too much, as it can open or close the panel. We probably need a simple `closeConditionalPanel` and it should be idempotent. In other words, regardless of what the state of the conditional panel is, the result should be a closed panel. this would get rid of the else if case and would make the code more precise. We should move away from using `toggle`s in any case, unless it is in very special circumstances, since it is too ambiguous for most use cases. 

Doing it in a reducer is a good idea and would likely be simplest to implement. Actions should describe what happened, they shouldn't be used as setters. I don't know if we have a reducer for this logic, but if we do, that could be a clean way of ensuring that it is always closed.